persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SampleSparkRepository.java

```diff
@@ -0,0 +1,197 @@
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Sample getSampleInStudy(String studyId, String sampleId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getAllSamplesOfPatientInStudy(String studyId, String patientId, String projection,
+                                                      Integer pageSize, Integer pageNumber, String sortBy,
+                                                      String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSamplesOfPatientInStudy(String studyId, String patientId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getAllSamplesOfPatientsInStudy(String studyId, List<String> patientIds, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getSamplesOfPatientsInMultipleStudies(List<String> studyIds, List<String> patientIds,
+                                                              String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> fetchSamples(List<String> studyIds, List<String> sampleIds, String projection) {
+
+        Dataset<Row> samples = loadStudyFiles(new HashSet<>(studyIds), ParquetConstants.CLINICAL_SAMPLE);
+        
+        if (!CollectionUtils.isEmpty(sampleIds)) {
+            samples = samples.where(samples.col("SAMPLE_ID").isin(sampleIds.toArray()));
+        }
+        
+        samples.createOrReplaceTempView("samples");
+        StringBuilder sb = new StringBuilder("SELECT PATIENT_ID, SAMPLE_ID, studyId");
+        if (PersistenceConstants.SUMMARY_PROJECTION.equalsIgnoreCase(projection)
+            || PersistenceConstants.DETAILED_PROJECTION.equalsIgnoreCase(projection)) {
+            sb.append(", SAMPLE_TYPE");
+        }
+        sb.append(" FROM samples");
+
+        Dataset<Row> res = spark.sql(sb.toString());
+        
+        return res.collectAsList().stream().
+            map(r -> mapToSample(r, projection)).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Sample> fetchSamples(List<String> sampleListIds, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaSamples(List<String> studyIds, List<String> sampleIds) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaSamples(List<String> sampleListIds) {
+        throw new UnsupportedOperationException();    }
+
+    @Override
+    public List<Sample> getSamplesByInternalIds(List<Integer> internalIds) {
+        throw new UnsupportedOperationException();
+    }
+
+    private Dataset<Row> loadStudyFiles(Set<String> studyIds, String file) {
```

Comment:
looks similar to the one in `ClinicalDataSparkRepository.java`. may be better to move it to utils class.

Reply:
Agreed. I added a util class.